### ISSUE TITLE
Why train generator using real_labels in GAN?

### ISSUE THREAD

**Issue by jamestang0219** [yunjey/pytorch-tutorial#45]
Hello, 
I'm confused about the codes of training generator in GAN. Anyone could help me?
Codes here:
```
        #=============== Train the generator ===============#
        # Compute loss with fake images
        z = to_var(torch.randn(batch_size, 64))
        fake_images = G(z)
        outputs = D(fake_images)
        g_loss = criterion(outputs, real_labels)
        
        # Backprop + Optimize
        D.zero_grad()
        G.zero_grad()
        g_loss.backward()
        g_optimizer.step()
```
Images generated from generator are fake, but why the second parameter of criterion is real_labels not fake_labels? Thank you all.

**Comment by AceCoooool** [comment]
Because the "task" of generator is to cheat the Discriminator. ~ How can you cheat the Discriminator ? ---> you can let the Generator to generate "real image" ! , so the output of Generator is close to real_labels as possible. (If you use fake_label, you want to creat a bad image to cheat Discriminator ? )

**Comment by jamestang0219** [comment]
Thank you for your help @AceCoooool . In my opinion, training generator won't update discriminator's parameters at the same time, so if the image from generator is fake enough, the loss of criterion will be huge. Using this loss to update generator's parameters, they will be close to real image. Thank you again, :)
